### PR TITLE
Enhance docs contributor info (2.16.4 backport from main)

### PR DESCRIPTION
**Pull request by michelle-purcell**
This PR contains content changes for the doc contributor guide that were SME approved and merged to `main` as per:

- PR #31127 
- PR #30586

### CODE REVIEW

**Comment by michelle-purcell** [comment]
@gsmet - FYI. These doc contributor guide enhancements are not specific to 2.16.x or 3.x versions.
Please publish these changes with 2.16.4 or the next opportunity to refresh the [docs portal](https://quarkus.io/guides/). Many thanks.

**Review by gastaldi — APPROVED** [review]

File: docs/src/main/asciidoc/doc-create-tutorial.adoc [hunk 1]
```diff
@@ -180,7 +180,7 @@ Let's add the following code to that file:
 // tag::application[]
 package org.acme;
 
-import javax.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.context.ApplicationScoped;
```
Comment (gastaldi) [inline docs/src/main/asciidoc/doc-create-tutorial.adoc:183]: That is a 3.x content and won't work in 2.16.x

Comment (michelle-purcell) [reply docs/src/main/asciidoc/doc-create-tutorial.adoc]: @gastaldi & @gsmet - Thanks :+1: Fixed.

**Comment by michelle-purcell** [comment]
Thanks @gsmet.

**Pull request merged**